src/coreclr/vm/interopconverter.cpp

```diff
@@ -189,17 +189,19 @@ IUnknown *GetComIPFromObjectRef(OBJECTREF *poref, ComIpType ReqIpType, ComIpType
         {
             hr = SafeQueryInterface(pUnk, IID_IDispatch, &pvObj);
```

Comment:
The logic here is still a bit off. The input is likely going to be `ComIpType_Both` &ndash; which means check for both. Which means at this point we are sure it is at least `ComIpType_Unknown` because `TryGetComIPFromObjectRefUsingComWrappers` returned `true`. In this branch we QI for `IDispatch` if `ComIpType_Dispatch` is set. If the QI is successful we can return `ComIpType_Dispatch` otherwise we simply fallback to `ComIpType_Unknown`.

Reply:
@AaronRobinsonMSFT thanks. Fixed.